hydrocarbon_problem/env/env.py

```diff
@@ -46,97 +62,186 @@ def __init__(self, initial_feed_spec: StreamSpecification = DEFAULT_INITIAL_FEED
         self._current_stream_number = self._initial_feed.number
         self._steps = 0
 
-    
-    def observation_spec(self):
-        pass
-
-    def action_spec(self):
-        pass
+        self._blank_state = np.zeros(self.observation_spec().shape)
 
-    def reward_spec(self):
-        pass
+    
+    def observation_spec(self) -> specs.Array:
+        input_obs = self._stream_to_observation(self._initial_feed)
+        return specs.Array(shape=input_obs.shape, dtype=float)
 
     def discount_spec(self):
-        pass
+        discount_tops = specs.BoundedArray(
+            shape=(), dtype=float, minimum=0., maximum=1., name='discount_tops')
+        discount_bots = specs.BoundedArray(
+            shape=(), dtype=float, minimum=0., maximum=1., name='discount_bots')
+        overall = specs.BoundedArray(
+            shape=(), dtype=float, minimum=0., maximum=1., name='discount_overall')
+        return Discount(
+            overall=overall,
+            created_states=(discount_tops, discount_bots))
+
+    def action_spec(self) -> Tuple[specs.DiscreteArray, specs.BoundedArray]:
+        continuous_spec = specs.BoundedArray(shape=(5,), dtype=float, minimum=-1, maximum=1,
+                                  name="action_continuous")
```

Comment:
The shape is 5, because we have 5 RadFrac parameters which we set?

Reply:
Yes